src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -523,82 +524,57 @@ QString parseDeviceDB(mixxx::DbConnectionPoolPtr dbConnectionPool, TreeItem* dev
     bool folderOrPlaylistFound = false;
 
     for (int tableOrderIndex = 0; tableOrderIndex < totalTables; tableOrderIndex++) {
-        for (std::vector<rekordbox_pdb_t::table_t*>::iterator table =
-                        reckordboxDB.tables()->begin();
-                table != reckordboxDB.tables()->end();
-                ++table) {
-            if ((*table)->type() == tableOrder[tableOrderIndex]) {
-                uint16_t lastIndex = (*table)->last_page()->index();
-                rekordbox_pdb_t::page_ref_t* currentRef = (*table)->first_page();
+        for (const auto& table : *rekordboxDB.tables()) {
+            if (table->type() == tableOrder[tableOrderIndex]) {
+                uint16_t lastIndex = table->last_page()->index();
+                rekordbox_pdb_t::page_ref_t* currentRef = table->first_page();
```

Comment:
this didn't build anymore because the `table_t` changed to a unique_ptr. I took the liberty and modernized the loop in the same step (same below).